lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -322,6 +348,53 @@ hdPassphrase (Key rootXPub) = Passphrase $
     (unXPub rootXPub)
     ("address-hashing" :: ByteString)
 
+toAddress
+    :: Key 'RootK XPub
+    -> Key 'AddressK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> Address
+toAddress rootXPub (Key addrXPub) accIx addrIx =
```

Comment:
Transformation of key to addresses should happen via the `keyToAddress` interface:

https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs#L397-L402

This however will requires a different definition for the `Key` type for random addresses. This is highlighted in #557 and also, justifies why #558 is actually blocked by #557. We need to make the wallet layer parametric over the the key types in order to be able to use different `Key` representation between the sequential and random schemes.

Reply:
removed `toAddress`